BenchmarkTests/Runner/Scenarios/Trace/UI/TraceContentView.swift

```diff
@@ -0,0 +1,173 @@
+                            .textFieldStyle(RoundedBorderTextFieldStyle())
+                            .frame(width: 80)
+                            .keyboardType(.numberPad)
+                        Stepper("", value: $depth, in: 1 ... 100, step: 1)
+                            .frame(width: 80)
+                    }
+
+                    HStack {
+                        Text("Child delay:")
+                        Spacer()
+                        TextField("Child delay:", value: $childDelay, formatter: NumberFormatter())
+                            .textFieldStyle(RoundedBorderTextFieldStyle())
+                            .frame(width: 80)
+                            .keyboardType(.numberPad)
+                        Stepper("", value: $childDelay, in: 1 ... 100, step: 1)
+                            .frame(width: 80)
+                    }
+                }
+
+                Button(action: sendTrace) {
+                    Text("Send")
+                        .foregroundColor(.white)
+                        .padding()
+                        .frame(maxWidth: .infinity)
+                        .background(Color.purple)
+                        .cornerRadius(8)
+                }
+                .listRowBackground(EmptyView())
+                .listRowInsets(EdgeInsets())
+
+                VStack(alignment: .center) {
+                    ProgressView()
+                        .progressViewStyle(CircularProgressViewStyle())
+                        .opacity(isSending ? 1 : 0)
+                        .padding()
+                    Text("Traces sent: \(traceCount)")
+                        .padding(.bottom, 16)
+                }
+                .frame(maxWidth: .infinity, alignment: .center)
+                .listRowBackground(EmptyView())
+                .listRowInsets(EdgeInsets())
+            }
+        }
+    }
+
+    /// Starts the tracing process based on the current configuration:
+    /// 1. Creates a root span. with the given configuration.
+    /// 2. Recursively generates child spans via `sendSpanTree`.
+    /// 3. Finishes the root span and updates the trace counter.
+    private func sendTrace() {
+        DispatchQueue.main.async {
+            isSending = true
+        }
+
+        queue1.async { [self] in
+            let rootSpan = tracer.startSpan(operationName: operationName)
+            rootSpan.setTag(key: SpanTags.resource, value: resourceName)
+
+            if isError {
+                rootSpan.log(
+                    fields: [
+                        OTLogFields.event: "error",
+                        OTLogFields.errorKind: "Simulated error",
+                        OTLogFields.message: "Describe what happened",
+                        OTLogFields.stack: "Foo.swift:42",
+                    ]
+                )
+            }
+
+            wait(seconds: 0.5)
+
+            queue2.sync {
+                sendSpanTree(parent: rootSpan, currentLevel: 0, maxDepth: depth)
+            }
```

Comment:
/suggestion No need to jump on a secondary queue, you can create the children on the same thread.